docs/detections/rules-ui-monitor.asciidoc

```diff
@@ -41,3 +41,51 @@ For example, if an event occurred at 10:01, and due to a 9-minute ingestion pipe
 
 [role="screenshot"]
 image::images/timestamp-override.png[]
+
+[float]
+[[ml-job-compatibility]]
+=== Troubleshoot missing alerts for {ml} jobs
+
+The <<prebuilt-ml-jobs,prebuilt {ml} jobs>> have dependencies on data fields
+that are populated by {beats} and {agent} integrations. In version 7.11, new
+{ml} jobs (<<security-linux-jobs>> and <<security-windows-jobs>>) were provided,
+which operate on newer ECS fields than the previous
+<<security-winlogbeat-jobs>> and <<security-auditbeat-jobs>> jobs. However, the
+<<prebuilt-rules,prebuilt rules>> were not updated to use the new {ml} jobs.
+Therefore, if you have upgraded your cluster and all of your {beats} and {agent} 
+integrations to 7.11 or later, you must update your existing rules to use the
+new jobs. If are still using earlier versions of {beats} and {agent} 
+integrations, however, you must create new rules that use the new {ml} jobs and 
+enable them to run alongside your existing rules.
```

Comment:
Is this accurate @randomuserid? My understanding was pre 7.11 beats/agent+rules+jobs were compatible. It's just that if you update your beats/agents different fields will be populated for which the new jobs are necessary. So as a result of a beats/agent upgrade, existing jobs/rules are incompatible, so you would need to upgrade to the new jobs/rules(manually). 

In this case though, do the new jobs work with earlier (pre-7.11) versions of the beats?

Reply:
The new jobs use fields that did not exist when the first batch were made in 2019. So the new jobs work with current events, and the old jobs work with older events, with some exceptions as fields change.

Reply:
So does this mean it would be helpful to clarify this paragraph with something like this?:

> Therefore if you have upgraded your cluster and all of your Beats and Agent integrations to 7.11 or later, you must stop the Security:Auditbeat and Security:Winlogbeat jobs and use the Security:Linux and Security:Windows jobs instead. In this case, you must update your existing rules to use these new jobs. If you are still using earlier versions of Beats and Agent integrations, however, you can use both the old and new jobs and just stop a subset of the old jobs to avoid duplication (e.g. see https://www.elastic.co/guide/en/security/master/prebuilt-ml-jobs.html#_footnotedef_1). In this case, you must create new rules that use the new jobs and enable them to run alongside your existing rules.

Reply:
It's complex enough to be hard to explain. As a user, I like seeing sets of logical options side by side in order to evaluate which set of conditions match my local conditions - maybe something like this;

- If running a fleet of Beats older then version 7.10, continue using the first generation ML jobs and rules that have been in the Security app since version 7.5.
- If running a fleet of Beats of version 7.10 or higher, use the second generation "v2" ML jobs, and optionally make clones of the existing ML rules, connected to the v2 jobs, in order to alert on anomalies from the v2 jobs.
- If running a mixed fleet of old and new Beats, or running a fleet of Beats and Elastic Endpoint agents, follow both of the previous steps to enable and use both the first and second generation ML jobs and rules.
- If running a non-Elastic data shipper to gather ECS-compatible Windows events, use the second generation "v2" ML jobs, and optionally make clones of the existing ML rules, connected to the v2 jobs, in order to alert on anomalies from the v2 jobs.

Reply:
Thanks for the details!  I've added that list to the doc, though I changed "7.10 or higher" to "7.11 or higher", since I don't see the Security:Windows or Security:Linux modules in 7.10 (https://github.com/elastic/kibana/tree/7.10/x-pack/plugins/ml/server/models/data_recognizer/modules).